pkg/connector/event_filters.go

```diff
@@ -0,0 +1,133 @@
+		EventHandler: func(event *oktaSDK.LogEvent, targetMap map[string][]*oktaSDK.LogTarget, rv *v2.Event) error {
+			if len(targetMap["UserGroup"]) != 1 {
+				return fmt.Errorf("okta-connectorv2: expected 1 UserGroup target, got %d", len(targetMap["UserGroup"]))
+			}
+			userGroup := targetMap["UserGroup"][0]
+			rv.Event = &v2.Event_ResourceChangeEvent{
+				ResourceChangeEvent: &v2.ResourceChangeEvent{
+					ResourceId: &v2.ResourceId{
+						ResourceType: resourceTypeGroup.Id,
+						Resource:     userGroup.Id,
+					},
+				},
+			}
+			return nil
+		},
+	}
+	ApplicationLifecycleFilter = EventFilter{
+		EventTypes:  mapset.NewSet[string]("app.lifecycle.create", "app.lifecycle.delete", "application.lifecycle.update"),
+		TargetTypes: mapset.NewSet[string]("AppInstance"),
+		EventHandler: func(event *oktaSDK.LogEvent, targetMap map[string][]*oktaSDK.LogTarget, rv *v2.Event) error {
+			if len(targetMap["AppInstance"]) != 1 {
+				return fmt.Errorf("okta-connectorv2: expected 1 AppInstance target, got %d", len(targetMap["AppInstance"]))
+			}
+			appInstance := targetMap["AppInstance"][0]
+			rv.Event = &v2.Event_ResourceChangeEvent{
+				ResourceChangeEvent: &v2.ResourceChangeEvent{
+					ResourceId: &v2.ResourceId{
+						ResourceType: resourceTypeApp.Id,
+						Resource:     appInstance.Id,
+					},
+				},
+			}
+			return nil
+		},
+	}
+	ApplicationMembershipFilter = EventFilter{
+		EventTypes:  mapset.NewSet[string]("application.user_membership.add", "application.user_membership.remove", "application.user_membership.update"),
+		TargetTypes: mapset.NewSet[string]("AppInstance"),
+		EventHandler: func(event *oktaSDK.LogEvent, targetMap map[string][]*oktaSDK.LogTarget, rv *v2.Event) error {
+			if len(targetMap["AppInstance"]) != 1 {
+				return fmt.Errorf("okta-connectorv2: expected 1 AppInstance target, got %d", len(targetMap["AppInstance"]))
+			}
+			appInstance := targetMap["AppInstance"][0]
+			rv.Event = &v2.Event_ResourceChangeEvent{
+				ResourceChangeEvent: &v2.ResourceChangeEvent{
+					ResourceId: &v2.ResourceId{
+						ResourceType: resourceTypeApp.Id,
+						Resource:     appInstance.Id,
+					},
+				},
+			}
+			return nil
+		},
+	}
+	RoleMembershipFilter = EventFilter{
+		EventTypes:  mapset.NewSet[string]("user.account.privilege.grant", "user.account.privilege.revoke"),
+		TargetTypes: mapset.NewSet[string]("ROLE", "User"),
+		EventHandler: func(event *oktaSDK.LogEvent, targetMap map[string][]*oktaSDK.LogTarget, rv *v2.Event) error {
+			if len(targetMap["ROLE"]) != 1 {
+				return fmt.Errorf("okta-connectorv2: expected 1 ROLE target, got %d", len(targetMap["ROLE"]))
+			}
+			role := targetMap["ROLE"][0]
+
+			// for some reason we don't get the role ID (or type) formatted properly.
+			// hack to look it up via DisplayName
+			roleType := StandardRoleTypeFromLabel(role.DisplayName)
+			if roleType == nil {
+				return fmt.Errorf("okta-connectorv2: expected 1 ROLE target, got %d", len(targetMap["ROLE"]))
```

Comment:
@mj-palanker I feel like the wording of this error should be tweaked to distinguish it from the error on line 76.